compiler/z/codegen/OpMemToMem.cpp

```diff
@@ -1517,18 +1527,19 @@ MemCmpVarLenMacroOp::generate(TR::Register* dstReg, TR::Register* srcReg, TR::Re
    TR::Compilation *comp = _cg->comp();
 
    if(cursorBefore == NULL) cursorBefore = _cg->getAppendInstruction();
+   generateShortCircuitInstructions(_cg, _rootNode, _resultReg, _srcReg, _dstReg, _doneLabel);
    generateLoop();
    setInRemainder(true);
    generateRemainder();
 
    _cursor = generateCmpResult(_cg, _rootNode, _dstReg, _srcReg, _resultReg, _falseLabel, _trueLabel, _doneLabel);
    TR::RegisterDependencyConditions * dependencies = generateDependencies();
    _cursor->setDependencyConditions(dependencies);
-   if(_startControlFlow==NULL)
-     {
-     _startControlFlow=cursorBefore->getNext();
-     if(_startControlFlow->getOpCodeValue() == TR::InstOpCode::assocreg) _startControlFlow=_startControlFlow->getNext();
-     }
+
+   _startControlFlow=cursorBefore->getNext();
```

Comment:
I think this could be the one place where we should be  refactoring the code. Couple of things seems wrong to me. 

1. Internal Control Flow starts with the BRC instruction you are generating for short circuit. So it should now be simple to have startICF label generated, and we do not need to go on after generating all instructions to find startICF.
2. I do not think we need to attach post dependency conditions to the start of internal control flow. Your snippet from instruction selection log suggests that we can fix that.

Reply:
Thanks your review. I start apply this change right now.